pkgs/tools/admin/azure-cli/default.nix

```diff
@@ -187,7 +246,12 @@ py.pkgs.toPythonApplication (py.pkgs.buildAzureCliPackage {
   # it's just a shebang script which calls `python -m azure.cli "$@"`
   postFixup = ''
     wrapProgram $out/bin/az \
-      --set PYTHONPATH $PYTHONPATH
+  '' + lib.optionalString withImmutableConfig ''
+    --set AZURE_IMMUTABLE_DIR $out/etc/azure \
+  '' + lib.optionalString (withExtensions != [ ]) ''
+    --set AZURE_EXTENSION_DIR ${extensionDir} \
+  '' + ''
+    --set PYTHONPATH $PYTHONPATH
```

Comment:
This leaks inputs from all phases, if there any. Better to use the better functions.

Reply:
Yes, this leaks things written in checkPhase into the output, good point. Not sure how to fix it tho.

> Better to use the better functions.

What function?

Reply:
Depending on what we need, we can either just prefix python.sitePackages https://github.com/NixOS/nixpkgs/blob/beea7e8ff8b7a3de32b9fe9d7473d99b5c1da503/pkgs/applications/audio/osdlyrics/default.nix#L75 , use withPackages https://github.com/NixOS/nixpkgs/blob/beea7e8ff8b7a3de32b9fe9d7473d99b5c1da503/nixos/modules/services/continuous-integration/buildbot/master.nix#L273 or make pythonPath https://github.com/NixOS/nixpkgs/blob/beea7e8ff8b7a3de32b9fe9d7473d99b5c1da503/nixos/modules/services/misc/tandoor-recipes.nix#L133 

I didn't look into what is more correct and the best solution here though

Reply:
Thanks, I've put this into a separate commit as it was somewhat unreleated to the other changes made.